### PR TITLE
Added all services and DAOs

### PR DESCRIPTION
**Pull request by TatianaPirch**

### CODE REVIEW

File: src/main/java/mate/academy/spring/dao/impl/LibraryDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        Rent rent = query.getSingleResult();
+        rent.setActive(false);
+        sessionFactory.getCurrentSession().update(rent);
+        return rent;
+    }
+
+    @Override
+    public List<Book> getBooksRentByUser(User user) {
+        TypedQuery<Rent> query = sessionFactory.getCurrentSession()
+                .createQuery("FROM Rent WHERE user =:user"
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/dao/impl/LibraryDaoImpl.java]:
```suggestion
                .createQuery("SELECT rent.book FROM Rent WHERE user =:user"
```

File: src/main/java/mate/academy/spring/entity/Book.java [hunk 2]
```diff
@@ -19,9 +25,22 @@
     private Integer year;
     private Double price;
 
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "authors_books",
+            joinColumns = @JoinColumn(name = "book_id"),
+            inverseJoinColumns = @JoinColumn(name = "author_id"))
+    private List<Author> authors;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/entity/Book.java]: should we use `join table` on the both side?

File: src/main/java/mate/academy/spring/service/impl/AuthorServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import mate.academy.spring.service.AuthorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class AuthorServiceImpl implements AuthorService {
+
+    @Autowired
+    AuthorDao authorDao;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/service/impl/AuthorServiceImpl.java]: add access modifier

File: src/main/java/mate/academy/spring/dao/impl/LibraryDaoImpl.java (hunk 1 above)
Comment (TatianaPirch) [reply src/main/java/mate/academy/spring/dao/impl/LibraryDaoImpl.java]: I modified query, but when I tried to use rent.book I got error : Unable to resolve path [rent.book], unexpected token [rent]

Comment (Sofasmile) [reply src/main/java/mate/academy/spring/dao/impl/LibraryDaoImpl.java]: ок)

**Pull request merged**